### PR TITLE
AdminStorage interface and API protos

### PR DESCRIPTION
**Pull request by codingllama**
Adds the AdminStorage interface, read-only and read/write AdminTX and the
corresponding Tree proto to the API.

### CODE REVIEW

File: trillian.proto [hunk 1]
```diff
@@ -22,12 +34,107 @@ enum SignatureAlgorithm {
...
+  // Optional.
+  string description = 9;
+
+  // Timestamp of tree creation.
+  // Readonly.
+  int64 create_time_millis_since_epoch = 10;
+
+  // Timestamp of last tree update.
+  // Readonly (automatically assigned on updates).
+  int64 update_time_millis_since_epoch = 11;
```
Comment (AlCutter) [inline trillian.proto]: Would it be useful to have a `repeated Audit` field where the API appends details of any mutations made to this pb?

Comment (codingllama) [inline trillian.proto]: Nice? Yes. Useful? Not so much, I think.

We only allow changes to display_name and description, which are pretty inconsequential ATM.

**Comment by codingllama** [comment]
Thanks for the quick review!

I pushed a few other changes, PTAL?

File: trillian.proto (hunk 1 above)
Comment (AlCutter) [inline trillian.proto]: I was thinking of when state could be changed in particular.

Comment (codingllama) [inline trillian.proto]: I wasn't planning on adding a history at this stage. I'd like to give it more thought if that's something we want to have.

**Pull request merged**